rotkehlchen/data/importer.py

```diff
@@ -1014,3 +1015,167 @@ def import_shapeshift_trades_csv(self, filepath: Path) -> Tuple[bool, str]:
                 except KeyError as e:
                     return False, str(e)
         return True, ''
+
+    def _consume_uphold_transaction(self, csv_row: Dict[str, Any]) -> None:
+        """
+        Consume the file containing both trades and transactions from uphold.
+        """
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['Date'],
+            formatstr='%a %b %d %Y %H:%M:%S %Z%z',
+            location='uphold',
+        )
+        destination = csv_row['Destination']
+        destination_asset = symbol_to_asset_or_token(csv_row['Destination Currency'])
+        destination_amount = deserialize_asset_amount(csv_row['Destination Amount'])
+        origin = csv_row['Origin']
+        origin_asset = symbol_to_asset_or_token(csv_row['Origin Currency'])
+        origin_amount = deserialize_asset_amount(csv_row['Origin Amount'])
+        fee = deserialize_fee(csv_row['Fee Amount']) or FVal('0.0')
+        fee_asset = symbol_to_asset_or_token(csv_row['Fee Currency'] or csv_row['Origin Currency'])
+        transaction_type = csv_row['Type']
+        notes = f"""
+Activity from uphold with uphold transaction id:
+ {csv_row['Id']}, origin: {csv_row['Origin']},
+ and destination: {csv_row['Destination']}.
+  Type: {csv_row['Type']}.
+  Status: {csv_row['Status']}.
+"""
+        # On Exchange Transactions and Trades.
+        if origin == destination == 'uphold':
+            # Transfers
+            if origin_asset == destination_asset:
+                # Transfers have the same amounts
+                if origin_amount == destination_amount:
```

Comment:
since I was switching on transaction_type, I found 3 values in my csv "in", "out", "transfer".  The else statement logs and unexpected "transfer" entry.  I do not believe any records will match `origin_asset == destination_asset` having `transaction_type == 'transfer'`